enterprise/frontend/src/embedding-sdk/store/auth/auth.ts

```diff
@@ -165,9 +185,23 @@ const sessionSchema = Yup.object({
   // as we don't use them, so we don't throw an error if they are missing
 });
 
-async function connectToInstanceAuthSso(url: string) {
+async function connectToInstanceAuthSso(
+  url: string,
+  authMethod?: MetabaseAuthMethod,
+) {
+  if (authMethod && authMethod !== "jwt" && authMethod !== "saml") {
+    throw MetabaseError.INVALID_AUTH_METHOD({
+      method: authMethod,
+    });
+  }
+
+  let ssoUrl = `${url}/auth/sso`;
+  if (authMethod) {
+    ssoUrl += `?preferred_method=${authMethod}`;
```

Comment:
Well, I'm not totally against this, but we have an established usage for `new URL()`. Try searching for that vs string concatenation on the URLs.

e.g.
https://github.com/metabase/metabase/blob/daea9c227659551ff820a3328701f14e4f79304b/docs/embedding/snippets/interactive-embedding-quick-start-guide/sso-with-jwt.ts#L85-L87

I'd prefer a more standard API for manipulating URL, just in case.

Reply:
Good idea, that looks nicer indeed. Lemme fix that

Reply:
Fixed in https://github.com/metabase/metabase/pull/58641/commits/7958df56ae34bfbc85647798c84a1a3111f2a7c8